include/deal.II/hp/refinement.h

```diff
@@ -159,36 +159,83 @@ namespace hp
 
     /**
      * Adapt the finite element on cells whose smoothness indicators meet a
-     * certain threshold.
+     * certain absolute threshold.
+     *
+     * For p-refinement and p-coarsening, two separate thresholds need to
+     * provided via parameters @p p_refine_threshold and @p p_coarsen_threshold.
+     *
+     * We consider a cell for p-adaptivity if it is flagged correspondingly and
+     * its indicator successfully compares to the corresponding threshold. Let
+     * us be more specific on the default case: We consider a cell for
+     * p-refinement if it is flagged for refinement and its indicator is larger
+     * than the corresponding threshold. The same applies for p-coarsening, but
+     * the cell's indicator must be lower than the threshold. However, different
+     * comparison functors can be supplied via the parameters @p compare_refine
+     * and @p compare_coarsen to impose different decision criteria.
+     *
+     * Each entry of the parameter @p indicators needs to correspond to an active
+     * cell.
+     *
+     * @note Preceeding calls of Triangulation::prepare_for_coarsening_and_refinement()
+     *   may change refine and coarsen flags, which will ultimately change the
+     *   results of this function.
+     */
+    template <int dim,
+              typename Number,
+              int spacedim,
+              typename ComparatorRefine  = std::greater<Number>,
+              typename ComparatorCoarsen = std::less<Number>>
+    void
+    p_adaptivity_from_absolute_threshold(
+      const hp::DoFHandler<dim, spacedim> &dof_handler,
+      const Vector<Number> &               indicators,
+      const Number                         p_refine_threshold,
+      const Number                         p_coarsen_threshold,
+      const ComparatorRefine &             compare_refine = ComparatorRefine(),
+      const ComparatorCoarsen &compare_coarsen = ComparatorCoarsen());
+
+    /**
+     * Adapt the finite element on cells whose indicators meet a certain
+     * threshold relative to their value range.
```

Comment:
In fact, those indicators could be any kind of criteria, since we basically require just a vector of numbers.

Instead of referring to `indicators` (which will be most commonly used for this function), we could call them `criteria` or `refinement criteria` similarly to ` GridRefinement::refine_and_coarsen_fixed_number`. I would prefer the general naming.